compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -144,6 +144,7 @@ class ScalaSettings extends Settings.SettingGroup {
   val YcheckAllPatmat: Setting[Boolean] = BooleanSetting("-Ycheck-all-patmat", "Check exhaustivity and redundancy of all pattern matching (used for testing the algorithm)")
   val YretainTrees: Setting[Boolean] = BooleanSetting("-Yretain-trees", "Retain trees for top-level classes, accessible from ClassSymbol#tree")
   val YshowTreeIds: Setting[Boolean] = BooleanSetting("-Yshow-tree-ids", "Uniquely tag all tree nodes in debugging output.")
+  val YnoOpaqueTypes: Setting[Boolean] = BooleanSetting("-Yno-opaque-types", "Opaque types are treated as ordinary transparent types.")
```

Comment:
I'm not sure what the underlying issue exactly is here (perhaps we just need to set better positions somewhere ?) but this is almost surely the wrong approach, we should avoid adding extra code path which are only used in very specific cases, every one of them doubles the number of possible states we need to test.

Reply:
Please see my reasoning and motivation in [this comment](https://github.com/lampepfl/dotty/pull/6414#issuecomment-488304296). I agree that it is a suboptimal approach, but it seems that we don't have the info required to target the underlying type properly on completion. Do you have any thoughts on how we could do so?

Reply:
This also leaks the opaque type
```
> opaque type T = Int => Int; (null: T).app
```
outocompletes the `apply` when it should not